scripts/wait-modelcar.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+if [ "${MODEL_INIT_MODE}" = "async" ] ; then
+  echo "Waiting for model files (modelcar) to be present..."
+  until test -e /mnt/models; do
+    sleep 1
+  done
+
+  echo "Model files are now available."
+fi
+
+echo "Starting model server..."
+eval $@
```

Comment:
* You should use `exec` to avoid spawning an extra shell, so saving up some resources.
* With using `eval` and if the tgi runtime creates child-process that are not properly reaped, you create zombie processes (not really an issue, as the pod will die in this case anyway)
* Also, `"$@"` should be used instead of `$@` to preserve the argument boundaries.
* No timeout so you might run forever

An improved version could look like:

```bash
#!/bin/bash

# Set a timeout of 300 seconds (5 minutes)
MAX_WAIT=300
WAIT_INTERVAL=1
TIME_WAITED=0

if [ "${MODEL_INIT_MODE}" = "async" ] ; then
  echo "Waiting for model files (modelcar) to be present..."
  while ! test -e /mnt/models; do
    sleep $WAIT_INTERVAL
    TIME_WAITED=$((TIME_WAITED + WAIT_INTERVAL))
    
    if [ "$TIME_WAITED" -ge "$MAX_WAIT" ]; then
      echo "Error: Timed out waiting for model files." >&2
      exit 1
    fi
  done
  echo "Model files are now available."
fi

echo "Starting model server..."
exec "$@"
```

Reply:
We no longer need this, but I'll take your feedback for future scripts :-)